### ISSUE TITLE
[declarations] doesn't work with builtins

### ISSUE THREAD

**Issue by LegNeato** [facebook/flow#6631]
`[declarations]` works with your own types or types defined in node modules, but does not suppress errors in flow builtins. See https://github.com/facebook/flow/pull/4916#issuecomment-404297827.

Repro in https://github.com/yamov/FlowDeclarations/.

**Comment by LegNeato** [comment]
Flow appears to parse all builtins into a global module. When using `Context.remove_all_errors cx` it doesn't do so for that module, and thus Flow throws errors for the builtins at some path like `/private/tmp/flow/flowlib_eca9b02/dom.js:823:13`. Adding `.*/tmp/.*` to `[declarations]` does not silence the errors as flow appears to treat builtins differently.

**Comment by wcandillon** [comment]
Thanks for filing this issue. I'm super excited about the use of `[declarations]` but unfortunately, I'm still getting a lot of type errors.

**Comment by LegNeato** [comment]
I've been debugging this a bit. I'm looking for a simple place to make the fix but right now it looks like the changes may have to be fairly invasive.

**Comment by nodkz** [comment]
@LegNeato @mroch I still think that `[declarations]` is the wrong name. It does not explain what it really means. Maybe `[supress]` is the better candidate?

For now for supressing wrong or stale declarations in packages i need to manage 2 files. Maybe it helps to find easy fix ;) All what is under `[supress]` became `{ declare module.exports: any; }`:

`.flowconfig`
```diff
- [ignore]
+ [supress]
.*/node_modules/react-native/Libraries/Lists/FlatList.js
.*/node_modules/react-native/Libraries/Lists/VirtualizedList.js
.*/node_modules/react-native/Libraries/Lists/SectionList.js
.*/node_modules/react-native/Libraries/CameraRoll/CameraRoll.js

.*/node_modules/react-native/Libraries/ReactNative/requireNativeComponent.js
.*/node_modules/react-native/Libraries/ReactNative/renderApplication.js
.*/node_modules/react-native/Libraries/Components/Switch/Switch.js
.*/node_modules/react-native/Libraries/Animated/src/nodes/AnimatedValue.js
.*/node_modules/react-native/Libraries/Animated/src/nodes/AnimatedInterpolation.js
```

`./flow-typed/fix-brocken-libdefs.js`
```diff
- declare module 'FlatList' { declare module.exports: any; }
- declare module 'VirtualizedList' { declare module.exports: any; }
- declare module 'SectionList' { declare module.exports: any; }
- declare module 'CameraRoll' { declare module.exports: any; }

- declare module 'renderApplication' { declare module.exports: any; }
- declare module 'requireNativeComponent' { declare module.exports: any; }
- declare module 'Switch' { declare module.exports: any; }
- declare module './AnimatedValue' { declare module.exports: any; }
- declare module './nodes/AnimatedValue' { declare module.exports: any; }
- declare module '../nodes/AnimatedValue' { declare module.exports: any; }
- declare module './AnimatedInterpolation' { declare module.exports: any; }
- declare module 'AnimatedInterpolation' { declare module.exports: any; }
- declare module './nodes/AnimatedInterpolation' { declare module.exports: any; }
```

PS. Please remove confusing `[declarations]` name in options.

**Comment by nodkz** [comment]
Small poll:
[declarations] - 👎
[silence] - 👍
[supress] - ❤️
[supress_as_any] - 😄
[untyped] - 😕
[relax] - 🎉

**Comment by wcandillon** [comment]
I actually like `[declarations]` 😅 @nodkz some of the names above don't describe what this option actually does. Like `[supress_as_any]` or `[untyped]`.

On a side note, I'm still not able to use this option correctly with React and React Native, am I missing something or this is a known issue?

**Comment by LegNeato** [comment]
@wcandillon Known issue: https://github.com/facebook/flow/issues/6631

**Comment by wcandillon** [comment]
@LegNeato Thanks a lot 🙋🏼‍♂️ Was just double checking 😉Really appreciate your support.

**Comment by lll000111** [comment]
@wcandillon 

> some of the names above don't describe what this option actually does

`[declarations]` _definitely_ doesn't.

**Comment by haggholm** [comment]
For the record, *suppress* is spelled with two *p*s. Other than that it’s IMO the most descriptive name, but hardly the focus of this issue…

**Comment by hjylewis** [comment]
Yeah if you want to change the name can you create a different issue, @nodkz?

**Comment by ericketts** [comment]
@LegNeato you mention the changes to get this working potentially being quite invasive, would you be willing to share some of what you discovered as to what the issue actually is and what might be needed to overcome it? I'm not terribly well versed in ocaml but would be happy to see if I can help if you'd be willing to get me up to speed on what you know so far.

**Comment by LegNeato** [comment]
@ericketts Essentially all I know is in https://github.com/facebook/flow/issues/6631#issuecomment-407280186. I haven't had time to dig in more and am almost exclusively using Rust these days so I don't personally have a pressing need (also I am an OCaml n00b). I basically just started at my landed changes and tried to follow the callers to see how builtins were handled. I didn't see an obvious place where builtin errors are handled differently than non-builtin, but they must be because `Context.remove_all_errors cx` in my diff doesn't clear their errors.

**Comment by jedwards1211** [comment]
@nodkz you forgot `[STFU]` 😄

**Comment by STRML** [comment]
Fixed in #8105 by my testing. Please try it out.
